src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -132,7 +226,18 @@ void PacketDataReporter::OnSrvRecord(SerializedQNameIterator name, const SrvReco
     }
 }
 
-void PacketDataReporter::OnIPAddress(const chip::Inet::IPAddress & addr)
+void PacketDataReporter::OnCommissionableNodeSrvRecord(SerializedQNameIterator name, const SrvRecord & srv)
+{
+    // Host name is the first part of the qname
+    mdns::Minimal::SerializedQNameIterator it = srv.GetName();
+    if (!it.Next())
+    {
+        return;
+    }
+    strncpy(mCommissionableNodeData.hostName, it.Value(), CommissionableNodeData::kHostNameSize);
```

Comment:
Let's not use `str*` anywhere `mdns::Minimal` API is used, because it return byte spans and DNS-SD and mDNS do not include any nul terminators in payloads.

Overall, only byte spans should be used, and treated as byte spans.

Reply:
It is a string (const char* really) though at this point- it's ascii cars, and the resolver delegate puts it into a const char* with a null terminator. It comes off the wire in a byte span, but gets transformed when it's put into the commissionable node data structure. Are you saying you want the structure to include the byte span directly?

Reply:
Just want to make sure nothing comes in that doesn't actually have a nul-terminator, If it's certain that all APIs being used provide nul terminator, that may work.

Also, `strncpy` is not guarateed to nul-terminate output, if input is >= CommissionableNodeData::kHostNameSize. Overall, suggest working primitives that ensure we get zero chances of not-nul-terminated data stored in memory pointed by any `const char *` or `char *`

Reply:
So I'm actually wary about the ByteSpan - it has a pointer, but doesn't actually own the data. In this case, the CommissionableNodeData structure owns the data itself and it can print directly. I can change this to use strcpy, but I do think the implementation is cleaner if the CommissionableNodeData structure has a proper copy of all the information it needs in the correct form.

Reply:
I didn't mean to lose ownership. I meant that when accessed, it would be good to use byte spans. Yes we need to have copies.